### PR TITLE
Added template tabo-hdinsight

### PR DESCRIPTION
**Pull request by mukulkgupta**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@mukulkgupta__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: tabo-hdinsight/Azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+          "tags": {
+            "displayName": "cse"
+          },
+          "properties": {
+            "publisher": "Microsoft.Compute",
+            "type": "CustomScriptExtension",
+            "typeHandlerVersion": "1.8",
+            "autoUpgradeMinorVersion": true,
+            "settings": {
+              "fileUris": [ "https://sapiotmlblob.blob.core.windows.net/hdinsightscript/sqlnew.ps1" ],
```
Comment (gatneil) [inline tabo-hdinsight/Azuredeploy.json]: Please see #2 in best practice checklist: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

File: tabo-hdinsight/Azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+              "direction": "Inbound",
+              "sourcePortRange": "*",
+              "destinationAddressPrefix": "*"
+            }
+          }
+        ]
+      }
+    },
+    {
+      "apiVersion": "[variables('TaboEhVersion')]",
```
Comment (gatneil) [inline tabo-hdinsight/Azuredeploy.json]: Please see #5 in best practice checklist: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

File: tabo-hdinsight/Azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+    "TaboVmdns": "[concat('tab', uniquestring(resourcegroup().id))]",
+    "TaboScriptfolder": ".",
+    "TaboScriptfilename": "sqlnew.ps1",
+    "TaboServiceBusNamespaceName": "[concat('sb', uniquestring(resourcegroup().id))]",
+    "TaboEhVersion": "2015-08-01",
+    "TaboDefaultSASKeyName": "RootManageSharedAccessKey",
+    "TaboDefaultAuthRuleResourceId": "[resourceId('Microsoft.ServiceBus/namespaces/authorizationRules', variables('TaboServiceBusNamespaceName'), variables('TaboDefaultSASKeyName'))]",
+    "TaboEventhubin": "IotInputxx",
+    "TaboEventhubout": "Iotoutputxx",
+    "TaboInKeyP": "DG4TcCs+XnDN83VZcT2e5NfVeSFvTuiYUiWtXhxnJ2s=",
```
Comment (gatneil) [inline tabo-hdinsight/Azuredeploy.json]: Are these actual keys with permissions to actual objects? If so, these should be replaced with placeholders. If replacing with placeholders, should probably make them parameters as well

File: tabo-hdinsight/Azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,555 @@
...
+    "TaboOutpolicy": "iotoutput",
+    "TaboStorageAccountNamex": "[concat('sto',uniquestring(resourceGroup().id))]",
+    "TaboDefaultApiVersion": "2015-05-01-preview",
+    "TaboClusterApiVersion": "2015-03-01-preview",
+    "TaboClustertype": "Spark",
+    "TaboClustername": "[concat('hd', uniquestring(resourcegroup().id))]",
+    "TaboClusterLoginUserName": "admin",
+    "TaboClusterLoginPassword": "Password@123",
+    "TaboSshUserName": "azureuser",
+    "TaboSshPassword": "Password@123",
```
Comment (gatneil) [inline tabo-hdinsight/Azuredeploy.json]: definitely shouldn't hard-code passwords and keys in this template; please make these parameters

File: tabo-hdinsight/Azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,555 @@
...
+    "TaboInKeyP": "DG4TcCs+XnDN83VZcT2e5NfVeSFvTuiYUiWtXhxnJ2s=",
+    "TaboInKeyS": "IAvcmGyxkAHoOvVRLrbuPRgGw+RqAa/KedmOaQYxTt8=",
+    "TaboOutKeyP": "Xo5cCWHgJh3Sag00I5BiG6PlfTHBuSsCWQOyj8wFdhU=",
+    "TaboOutKeyS": "ANYLcW/XZBhuuXfzaqwyUfxIa1HngNTGvgMyyjvnpj4=",
+    "taboServicebusSku": "standard",
+    "taboServiceBusApiVersion": "2015-08-01",
+    "TaboInpolicy": "iotinput",
+    "TaboOutpolicy": "iotoutput",
+    "TaboStorageAccountNamex": "[concat('sto',uniquestring(resourceGroup().id))]",
+    "TaboDefaultApiVersion": "2015-05-01-preview",
```
Comment (gatneil) [inline tabo-hdinsight/Azuredeploy.json]: is there a non-preview api version we can use?

File: tabo-hdinsight/Azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,555 @@
...
+    "TaboClustername": "[concat('hd', uniquestring(resourcegroup().id))]",
+    "TaboClusterLoginUserName": "admin",
+    "TaboClusterLoginPassword": "Password@123",
+    "TaboSshUserName": "azureuser",
+    "TaboSshPassword": "Password@123",
+    "TaboScriptActionUri": "https://sapiotmlblob.blob.core.windows.net/hdinsightscript/newScript.sh",
+    "TaboClusterWorkerNodeCount": 2,
+    "TaboSingleq": "'",
+    "TaboSpace": " ",
+    "TaboDbdns": "[concat('jdbc:sqlserver://',variables('TaboVmdns'),'.',variables('TaboLocation'),'.cloudapp.azure.com;database=IOT_Stream;user=sa;password=Dynapt@123')]",
```
Comment (gatneil) [inline tabo-hdinsight/Azuredeploy.json]: same

File: tabo-hdinsight/Azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,555 @@
...
+              },
+              "datasource": {
+                "type": "Microsoft.ServiceBus/EventHub",
+                "properties": {
+                  
+                  "serviceBusNamespace": "[variables('TaboServiceBusNamespaceName')]",
+                  "sharedAccessPolicyName": "[variables('TaboOutpolicy')]",
+                  "sharedAccessPolicyKey": "[variables('TaboOutKeyP')]",
+                  "eventHubName": "[variables('TaboEventhubout')]"
+
```
Comment (gatneil) [inline tabo-hdinsight/Azuredeploy.json]: Please remove extra whitespace

**Comment by gatneil** [comment]
Please address the above comments; also, please sign the CLA; otherwise I can't merge this PR :(

File: tabo-hdinsight/Azuredeploy.json (hunk 3 above)
Comment (mukulkgupta) [inline tabo-hdinsight/Azuredeploy.json]: I am working on replacing the Username & Passwords with Parameters, only issue being the Event Hub Keys that I have hard coded. How can we have the CI Bot generate unique keys. What function should be used there.

Comment (gatneil) [inline tabo-hdinsight/Azuredeploy.json]: What format are Event Hub Keys? The CI can generate ssh-format public keys (GEN-SSH-PUB-KEY), but at the moment that's it; it's okay if the PR doesn't pass CI due to this; we can still merge it in as long as you've tested it manually :)

File: tabo-hdinsight/Azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,555 @@
+{
```
Comment (singhkays) [inline tabo-hdinsight/Azuredeploy.json]: filename needs to be azuredeploy.json (note the case)

File: tabo-hdinsight/Azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,555 @@
...
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+  },
+  "variables": {
+    "TaboVnetId": "[resourceId('Microsoft.Network/virtualNetworks', variables('TaboVirtualNetworkName'))]",
```
Comment (singhkays) [inline tabo-hdinsight/Azuredeploy.json]: parameter names should follow camelCase convention

**Comment by gatneil** [comment]
no activity; closing

**Pull request closed without merging**